src/command/fetch.rs

```diff
@@ -21,6 +21,7 @@ pub(crate) struct Fetch {
 impl Command for Fetch {
     fn run(mut self, session: &mut Session) -> Fallible<ExitCode> {
         session.add_event_start(ActivityKind::Fetch);
+        ToolSpec::check_args(std::env::args(), "fetch".into())?;
```

Comment:
`std::env::args()` Will panic if any of the args is not valid UTF-8, so we've used `args_os()` everywhere else to deal with arguments. We should probably maintain that to prevent panics from malformed invocations.

Reply:
Ah, I had wondered about that. Will make that change!

Reply:
Ended up not needing to mess with `OsStr` at all, by way of refactoring it to do as you suggested and using `Vec<String>` for the various command structs.